src/core/thread/indirect_sender.cpp

```diff
@@ -180,13 +181,16 @@ void IndirectSender::SetChildUseShortAddress(Child &aChild, bool aUseShortAddres
 exit:
     return;
 }
+#endif
 
 void IndirectSender::HandleChildModeChange(Child &aChild, Mle::DeviceMode aOldMode)
 {
+#if OPENTHREAD_FTD
```

Comment:
```suggestion
```
Remove this macro where it is called

Reply:
Some explanation here: I directly reuse the code from `IndirectSender` and enable these code when `OPENTHREAD_FTD` **or** `OPENTHREAD_MTD_S2S` is enabled. However, unfortunately, there's still a little part of code that only makes sense for `FTD` in `IndirectSender`. In this `HandleChildModeChange` method, the short address logic is only for Child and doesn't make sense for the 'Peer' in S2S case.